### DISCUSSION TITLE
Displaying dynamic data in a react-table

### DISCUSSION THREAD

**Discussion by fre-ben** [TanStack/table#3395] [Q&A]
Hey!
I'm wondering if it is a bad idea to use react-table if I want to display data which changes with the user's input in a searchbar.
The reason I'm asking is, that the documentation advocates to use `useMemo` for the data.
As far as I understand `useMemo` this would hinder rerendering when the data changes.

Would it be bad practice to pass stateful data to the react-table?

Edit: Just had another idea. What if I pass stateful data to the react-table component and inside the component I would first "memoize" the stateful data and then use it like this:
```
  const { getTableProps, getTableBodyProps, headerGroups, rows, prepareRow } =
    useTable({ columns, data: memoizedData });
```
🤔

**Comment by Michael-Rimmer** [comment]
I'm not sure about memoizing the stateful data when it is passed in but the [editable data example ](https://react-table.tanstack.com/docs/examples/editable-data) seems to not bother with useMemo at all for the data (only for the columns). Although, this might be because it's using toy, sample data.
